code/ai/aicode.cpp

```diff
@@ -13892,29 +13912,41 @@ int maybe_big_ship_collide_recover_frame(object *objp, ai_info *aip)
 {
 	float	dot, dist;
 	vec3d	v2g;
+
+	bool better_collision_avoid_and_fighting = The_mission.ai_profile->flags[AI::Profile_Flags::Better_collision_avoidance] && aip->mode == AIM_CHASE;
+
+	// if this guy's in battle he doesn't have the time to spend up to 35 seconds recovering!
+	int phase1_time = better_collision_avoid_and_fighting ? 2 : 5;
```

Comment:
Might be good to expose this value at some point and make it settable via the ai table.  Maybe within the scope of this PR?

Reply:
I'm generally averse to adding profile options if the code can handle it properly in general for all cases (obviously exempting the very flag I'm adding), which is the intention of the values chosen here as well as with `collision_avoidance_aggression`. The only use case I can think of for exposing it would be to choose unusual values to fit unusual ships, which this *should* be able to handle, even as is. I am more than willing to re-examine this code to improve and generalize it further later, but once things go into the tables they rarely come out.